README.md

```diff
@@ -42,144 +76,321 @@ The only compile time dependency you will need is `okta-sdk-api`.  You will also
+```groovy
+compile 'com.okta.sdk:okta-sdk-api:${okta.version}'
+runtime 'com.okta.sdk:okta-sdk-impl:${okta.version}'
+runtime 'com.okta.sdk:okta-sdk-httpclient:${okta.version}'
 ```
 
-## Client configuration
+### SNAPSHOT Dependencies
 
-There are a few ways to configure the client, but the easiest way is to create a `~/.okta/okta.yaml`file and set the `token` and `orgUrl` values:
+Snapshots are deployed off of the 'master' branch to [OSSRH](https://oss.sonatype.org/) and can be consumed using the following repository configured for Apache Maven or Gradle:
 
-``` yaml
-okta:
-  client:
-    token: <your-api-token>
-    orgUrl: https://dev-123456.oktapreview.com
+```txt
+https://oss.sonatype.org/content/repositories/snapshots/
 ```
 
-## Creating a Client
+You'll also need:
 
- Once you create your `okta.yaml` file, you can create a client with a couple of lines:
+* An Okta account, called an _organization_ (sign up for a free [developer organization](https://developer.okta.com/signup) if you need one)
+* An [API token](https://developer.okta.com/docs/api/getting_started/getting_a_token)
+ 
+Construct a client instance by passing it your Okta domain name and API token:
+ 
+[//]: # (NOTE: code snippets in this README are updated automatically via a Maven plugin by running: mvn okta-code-snippet:snip.okta:okta-code-snippet-maven-plugin:snip)
+ 
+[//]: # (method: createClient)
+```java
+Client client = Clients.builder()
+    .setOrgUrl("https://dev-123456.oktapreview.com/")
+    .setClientCredentials(new TokenClientCredentials("your-api-token"))
+    .build();
+```
+[//]: # (end: createClient)
+ 
+Hard-coding the Okta domain and API token works for quick tests, but for real projects you should use a more secure way of storing these values (such as environment variables). This library supports a few different configuration sources, covered in the [configuration reference](#configuration-reference) section.
+ 
+## Usage guide
 
-``` java
-// Instantiate a builder for your Client. If needed, settings like Proxy and Caching can be defined here.
-ClientBuilder builder = Clients.builder();
+These examples will help you understand how to use this library. You can also browse the full [API reference documentation][javadocs].
 
-// No need to define anything else; build the Client instance. The ClientCredential information will be automatically found
-// in pre-defined locations.
-Client client = builder.build();
-```
+Once you initialize a `Client`, you can call methods to make requests to the Okta API.
 
-For more details see: [Creating a Client](https://github.com/okta/okta-sdk-java/wiki/Creating-a-Client)
+### Authenticate a User
 
-## Client CRUD Operations
+This library should be used with the Okta management API. For authentication, we recommend using an OAuth2 or OIDC library such as [Spring Security OAuth](https://spring.io/projects/spring-security-oauth) or [Okta's Spring Boot integration](https://github.com/okta/okta-spring-boot).
```

Comment:
We should point people to auth-sdk, like:

To call the Okta [Authentication API](https://developer.okta.com/docs/api/resources/authn), use the [Authentication SDK], or Okta's [Spring Boot integration].